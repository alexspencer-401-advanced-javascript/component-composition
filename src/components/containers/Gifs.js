import React, { Component } from 'react';
import Deck from '../Decks/Deck';
import { getGifs } from '../services/arnoldApi';

export default class Gif extends Component {
  state = {
    gifs: []
  }

  componentDidMount(){
    return getGifs()
      .then((fetchedGifs) => {
        this.setState({ 
          gifs: fetchedGifs 
        });
      });
  }

  render(){
    return (
      <Deck items={this.state.gifs}/>
    );
  }
}
