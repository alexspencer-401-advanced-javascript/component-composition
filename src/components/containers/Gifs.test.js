import React from 'react';
import { shallow } from 'enzyme';
import Gifs from './Gifs';


jest.mock('../services/arnoldApi.js', () => ({
  getGifs() {
    return Promise.resolve([
      {
        '_id': '5da23754845fd2cb76d59f10',
        'gifLink': 'https://media.tenor.com/images/7511947a6ae9acb62c9515bfba14b1ec/tenor.gif'
      }]);
  }
}));

describe('gif component', () => {
  it('makes a successful api call and returns gifs', () => {
    const wrapper = shallow(<Gifs />);
    expect(wrapper).toMatchSnapshot();
  });
});
