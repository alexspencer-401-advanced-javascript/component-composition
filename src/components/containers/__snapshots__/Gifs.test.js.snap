// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gif component makes a successful api call and returns gifs 1`] = `
<Deck
  props={Array []}
/>
`;
